### PR TITLE
NX-60 Add Die action

### PR DESCRIPTION
**Pull request by filip-szweda**
Changes:

- added 'Destroy' action
- it is checked whether 'Agent' / 'NPC' should be destroyed with every 'DealDmg' call
- 'Agent' / 'NPC' are destroyed if their 'hp' is at or below 0
- 'Agent' / 'NPC' are removed from the map when destroyed
- added necessary utils
- minor refactoring

### CODE REVIEW

File: src/scene.py [hunk 1]
```diff
@@ -144,3 +147,15 @@ def add_object_to_id_map(self, new_object: Object) -> None:
         Add object to per scene storage indexed by objects `id`.
         """
         self._objects_id_map[new_object.properties.id] = new_object
+
+    def decrement_objects_count(self) -> None:
+        self.objects_count -= 1
```
Comment (blazej-smorawski) [inline src/scene.py]: Breaks `ID` generator

**Review by Krystian030 — APPROVED** [review]
It look good!
Personally I think that separating the `die` method from the `Die` class is good because such a 
solution gives you the opportunity  to extend death activity for different "Creature", e.g. creating an NPC that will explode after death

File: src/scene.py [hunk 2]
```diff
@@ -144,3 +147,15 @@ def add_object_to_id_map(self, new_object: Object) -> None:
...
+
+    def remove_object_from_position_map(self, object_to_be_removed: Object):
+        self._objects_position_map[object_to_be_removed.properties.position].remove(object_to_be_removed)
```
Comment (blazej-smorawski) [inline src/scene.py]: Use `free_field`

File: src/scene.py [hunk 3]
```diff
@@ -144,3 +147,15 @@ def add_object_to_id_map(self, new_object: Object) -> None:
...
+
+    def remove_object_from_id_map(self, object_to_be_removed: Object):
+        for index, object in self._objects_id_map.items():
```
Comment (blazej-smorawski) [inline src/scene.py]: Use `properties.id`

File: src/objects/agent.py [hunk 4]
```diff
@@ -60,3 +61,8 @@ def tick(self) -> None:
             {'__builtins__': agent_builtins},
             self.scene.agent_locals,
         )
+
+    def die(self):
```
Comment (filip-szweda) [inline src/objects/agent.py]: Throw away this logic, just delete.

File: src/actions/die.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+from typing import Union
+
+
+from src.actions.action import Action
+from src.common.serializable import Properties
+from src.objects.npc import NPC
+
+
+class Die(Action):
```
Comment (filip-szweda) [inline src/actions/die.py]: Rename to `Destroy`

**Pull request merged**